aop/src/main/java/io/micronaut/aop/internal/intercepted/InterceptedMethodUtil.java

```diff
@@ -92,4 +93,46 @@ public static io.micronaut.core.annotation.AnnotationValue<?>[] resolveIntercept
         }
         return AnnotationUtil.ZERO_ANNOTATION_VALUES;
     }
+
+    /**
+     * Does the given metadata have AOP advice declared.
+     * @param annotationMetadata The annotation metadata
+     * @return True if it does
+     */
+    public static boolean hasAroundStereotype(AnnotationMetadata annotationMetadata) {
+        if (annotationMetadata == null) {
+            return false;
+        }
+        if (annotationMetadata.hasStereotype(Around.class)) {
+            return true;
+        } else if (annotationMetadata.hasStereotype(AnnotationUtil.ANN_INTERCEPTOR_BINDINGS)) {
+            return annotationMetadata.getAnnotationValuesByType(InterceptorBinding.class)
```

Comment:
If understand this logic correctly, the following code:

```java
return annotationMetadata.getAnnotationValuesByType(InterceptorBinding.class)
    .stream().anyMatch(av ->
        av.enumValue("kind", InterceptorKind.class).orElse(InterceptorKind.AROUND) == InterceptorKind.AROUND
);
```

if `annotationMetadata.getAnnotationValuesByType(InterceptorBinding.class)` returns a list of two items. 

- if the first item in the list returns for `av.enumValue("kind", InterceptorKind.class)` =  empty optional
- and the second item returns for `av.enumValue("kind", InterceptorKind.class)` == Optional(`INTRODUCTION`)

The above code will return `true`. Is that intended?

Should the above snippet be replaced by?

```java
Optional<InterceptorKind> interceptorKind = parseInterceptorKind(annotationMetadata.getAnnotationValuesByType(InterceptorBinding.class));
            if (!interceptorKind.isPresent()) {
                return true;
            }
            return interceptorKind.map(it -> it == InterceptorKind.AROUND).orElse(false);
...
}

    @NonNull
    private static Optional<InterceptorKind> parseInterceptorKind(@NonNull List<AnnotationValue<InterceptorBinding>> annotationValueList) {
        return annotationValueList
                .stream()
                .map(av -> av.enumValue("kind", InterceptorKind.class))
                .filter(Optional::isPresent) //TODO JDK 9 Min. Use .flatMap(Optional::stream())
                .map(Optional::get)
                .findFirst();
    }
```

Reply:
I didn't change this code as part of this PR, all I did was move to a common utility class. I don't think it makes sense to make changes in this area. But yes the default `InterceptorBinding` value if nothing is specified in `AROUND` so the code is correct and we shouldn't replace or change anything with this code.